Program.cs

```diff
@@ -8,33 +8,27 @@
 
 namespace myApp
 {
-    class Program
+    public class IrisData
     {
-        public class IrisData
-        {
-            [Column(0)]
-            public float SepalLength { get; set; }
-            [Column(1)]
-            public float SepalWidth { get; set; }
-            [Column(2)]
-            public float PetalLength { get; set; }
-            [Column(3)]
-            public float PetalWidth { get; set; }
-            [Column(4)]
-            public string Label { get; set; }
-        }
+        public float SepalLength { get; set; }
+        public float SepalWidth { get; set; }
+        public float PetalLength { get; set; }
+        public float PetalWidth { get; set; }
+        public string Label { get; set; }
+    }
 
-        public class IrisPrediction
-        {
-            public string PredictedLabel { get; set; }
-        }
+    public class IrisPrediction
+    {
+        public string PredictedLabel { get; set; }
+    }
 
+    class Program
+    {
         static void Main(string[] args)
         {
-            TextLoader loader = TextLoader.Create(typeof(IrisData));
-            loader.Separator = ",";
+            var loader = new TextLoader(typeof(IrisData), separator = ",");
```

Comment:
What syntax is this using?  If it's using named parameters, it would be `separator: ", "`, and if it's using property initializers, it'd be `new TextLoader(…) { separator = ", " }`

Reply:
Named parameters.

Reply:
> Named parameters.

That uses a colon then, not an equal sign.

Reply:
Yes sorry - no compiler to help

Reply:
@danmosemsft - The reason I used `TextLoader.Create` here is because the static `Create(Type)` method is implemented like:

```C#
static TextLoader Create(Type type)
{
    TextLoader loader = new TextLoader();
    
     foreach (Property p in type.GetProperties())
     {
          loader.Columns.Add(CreateColumn(p));
     }
     return loader;
}
```

The idea being that the default constructor is also `public`. So users can create a TextLoader, and start adding columns.  And alternatively we have a convenience method to create a TextLoader with a bunch of columns defaulted from the Type.

Reply:
Also, TextLoader has roughly 11 optional parameters. I don't think we want 11 optional parameters to the constructor. That's why the constructor takes all the required parameters, and any optional value (i.e. separator) is moved to a property.